src/encoding/predictProxyAddress.ts

```diff
@@ -9,6 +9,18 @@ import {
 
 import { address as factoryAddress } from "../factory/proxyFactory";
 
+/**
+ * Predicts the address of a proxy contract deployed using CREATE2.
+ *
+ * @param {Object} params - The function parameters.
+ * @param {string} [params.factory=factoryAddress] - The address of the factory contract.
+ * @param {string} params.mastercopy - The address of the mastercopy contract.
+ * @param {Object} params.setupArgs - The arguments for the setup function.
+ * @param {any[]} params.setupArgs.types - The types of the setup arguments.
+ * @param {any[]} params.setupArgs.values - The values of the setup arguments.
+ * @param {BigNumberish} params.saltNonce - The salt nonce used for CREATE2 deployment.
+ * @returns {string} The predicted address of the proxy contract.
+ */
```

Comment:
Proposal:
/**
 * Predicts the address of a proxy contract deployed via ModuleProxyFactory.
 *
 * @param {Object} params - The function parameters.
 * @param {string} [params.factory=factoryAddress] - The address of the factory contract.
 * @param {string} params.mastercopy - The address of the mastercopy contract.
 * @param {Object} params.setupArgs - The arguments for the setup function.
 * @param {any[]} params.setupArgs.types - The types of the setup arguments.
 * @param {any[]} params.setupArgs.values - The values of the setup arguments.
 * @param {BigNumberish} params.saltNonce - The saltNonce used to internally derive the final create2 salt.
 * 
 * @returns {string} The predicted address of the proxy contract.
 */